matrix_sdk/src/room/common.rs

```diff
@@ -169,13 +170,30 @@ impl Common {
     }
 
     async fn ensure_members(&self) -> Result<()> {
+        if !self.are_events_visible() {
+            return Err(
+                error::Error::Http(HttpError::NotClientRequest)
+            )
+        }
+
         if !self.are_members_synced() {
             self.request_members().await?;
         }
 
         Ok(())
     }
 
+    fn are_events_visible(&self) -> bool {
+        if let RoomType::Invited = self.inner.room_type() {
+            return match self.inner.history_visibility() {
+                HistoryVisibility::WorldReadable | HistoryVisibility::Shared => true,
```

Comment:
Did you test that `Shared` allows us to fetch the members? It doesn't seem like it should based on the spec:

> shared - Previous events are always accessible to newly joined members. All events in the room are accessible, even those sent when the member was not a part of the room.

We are not a newly joined member, we're still only invited.

On the other hand `Invited` might behave that way:

> invited - Events are accessible to newly joined members from the point they were invited onwards. Events stop being accessible when the member’s state changes to something other than invite or join.

We are invited, and should have access to the state.

Am I missing something here?

Reply:
Got it, I seem to have misunderstood the spec. Fixing.